ext/dapr-ext-workflow/dapr/ext/workflow/workflow_runtime.py

```diff
@@ -50,17 +52,30 @@ def __init__(self, host: Optional[str] = None, port: Optional[str] = None):
         self.__worker = worker.TaskHubGrpcWorker(host_address=uri.endpoint, metadata=metadata,
                                                  secure_channel=uri.tls)
 
-    def register_workflow(self, fn: Workflow):
+    def register_workflow(self, fn: Workflow, *, name: Optional[str] = None):
```

Comment:
Is it possible to add a small doc section somewhere in the python-sdk repo itself, where-in we briefly describe that re-registering (or mixing the registration methods to register multiple times) will not succeed?

Reply:
I had previously suggested that registering a workflow/activity with an explicit name argument that already has an assigned _alternate_name should raise an exception.

But now I wonder if there is a legitimate use-case for overriding a previously set `_alternate name`  during the registration step.  

Is it likely that activities and workflows might be packaged into a "python library", and the consumer of that library might want to override the alternate name during the registration process? 

Perhaps to have precise control over versioning so that existing workflows won't break?

Should the register function log a warning instead of raising an exception,  and then override the previously assigned `_alternate_name` with the passed in `name`?

I do think the decorator should raise an exception if `_alternate_name` is already set on the function. 

Finally, should the `_alternate_name` attribute be renamed to `_dapr_alternate_name`  just to namespace it a bit more?

Reply:
I think making the change from `_alternate_name` to`_dapr_alternate_name` is fine, I will make that change. 
On the other parts, I do not think for this PR we need to allow multiple registrations of a workflow with different names. Let's open a separate feature request and discuss the pros, cons of that feature.